sled-agent/src/opte/illumos/firewall_rules.rs

```diff
@@ -125,6 +125,7 @@ pub fn opte_firewall_rules(
     vni: &Vni,
     mac: &MacAddr6,
 ) -> Vec<FirewallRule> {
+    #[allow(clippy::map_flatten)]
```

Comment:
Justification: the structure of this whole loop is that we accumulate a doubly nested vector of translated rules (one for each host and protocol) and then flatten it twice to get a single vector that covers everything. If we wrote the flattening as `.flat_map(...).flatten()` as suggested by clippy, it would totally obscure the structure of the loop. It seems correct to replace _one_ level of `.map(...).flatten()` with `.flat_map(...)`, but (at least in this case) not two.

Reply:
Sounds reasonable.  I'd consider a comment explaining that (even if it's just "Clippy doesn't like `.map(...).flatten()` below, but in this case it's clearer than the alternative.").  Up to you.